//===== rAthena Patch ========================================
//= messagecolor
//===== By: ==================================================
//= AnnieRuru
//===== Current Version: =====================================
//= 0.3
//===== Compatible With: ===================================== 
//= rAthena 2019-02-15
//===== Description: =========================================
//= make the unit talk in colorize message
//===== Topic ================================================
//= https://rathena.org/board/topic/118229-npctalk-color-in-script/?do=findComment&comment=357625
//===== Additional Comments: =================================  
//= Note: this one use some trick to get the message display on self ... just not in color
//============================================================

 src/custom/script.inc     | 17 +++++++++++++++++
 src/custom/script_def.inc |  1 +
 2 files changed, 18 insertions(+)

diff --git a/src/custom/script.inc b/src/custom/script.inc
index 839b990cb..aef3a411c 100644
--- a/src/custom/script.inc
+++ b/src/custom/script.inc
@@ -17,3 +17,20 @@
 //	script_pushint(st,1);
 //	return 0;
 //}
+
+BUILDIN_FUNC(messagecolor) {
+	struct block_list *bl = map_id2bl( script_getnum(st,2) );
+	if (!bl) {
+		ShowWarning("buildin_messagecolor: Error in finding object GID %d!\n", script_getnum(st,2));
+		return SCRIPT_CMD_FAILURE;
+	}
+	if ( bl->type == BL_PC ) {
+		char msg[CHAT_SIZE_MAX];
+		safestrncpy( msg, script_getstr(st,3), CHAT_SIZE_MAX );
+		clif_messagecolor( bl, script_getnum(st,4), msg, true, AREA );
+		clif_showscript( bl, msg, SELF );
+	}
+	else
+		clif_messagecolor( bl, script_getnum(st,4), script_getstr(st,3), true, AREA);
+	return SCRIPT_CMD_SUCCESS;
+}
diff --git a/src/custom/script_def.inc b/src/custom/script_def.inc
index 886399273..ee4d969cd 100644
--- a/src/custom/script_def.inc
+++ b/src/custom/script_def.inc
@@ -9,3 +9,4 @@
  **/
 
 //BUILDIN_DEF(example,""),
+BUILDIN_DEF(messagecolor,"isi"),
